### DISCUSSION TITLE
Gpus all and ipc host

### DISCUSSION THREAD

**Discussion by chrisdengso** [dokku/dokku#7055] [Q&A]
Hi, 

I'm trying to deploy an app using Dockerfile. The docker container should have access to the host system those we want to run it with "--gpus all" and "ipc=host" flags. I've set those using dokku: 

```
=====> guard docker options information
       Docker options build:                                   
       Docker options deploy:                                  
       Docker options run:            --gpus all --ipc=host  
```

However it seems that the container does not have access to the Nvidia drivers on the host system. We're getting the error on deployment: `NVML Shared Library Not Found`.

The image we're running is `vllm/vllm-openai`, and running the 

Building and running the docker file manually works.

Any ideas?

**Comment by josegonzalez** [comment]
You almost certainly need to setup nvidia's docker drivers for GPU access.

**Reply by chrisdengso** [reply]
Thanks for having a look! We're using the [Docker image from vllm ](https://hub.docker.com/r/vllm/vllm-openai/) which should include the necessary drivers. We've also installed all the drivers and toolkits from Nvidia. 

We can run the Docker file manually with the given flags, and it boots up as expected. Could it be a bug or restriction with Dokku?

**Reply by josegonzalez** [reply]
Dokku doesn't do anything except execute `docker run` with your desired arguments (and maybe a few more). You can check the _exact_ command being run by enabling trace mode (`dokku trace:on` and deploy or `dokku --trace ps:rebuild $APP`). The `docker run` command should be towards the bottom (there's a bunch of debugging code).

My guess is that maybe you are missing a flag or you are running a specific container image that has a binary/drivers that the herokuish base image does not (assuming you are using herokuish).
